scripts/avatars/migrate_pids.py

```diff
@@ -0,0 +1,109 @@
+from great_ape_safe import GreatApeSafe
+from helpers.addresses import r
+
+# Here are exceptions that may exist during migration
+PID_AURA_EXCEPTION = 67
+PID_CONVEX_EXCEPTION = 0
+
+# New pids
+AURA_NEW_PID_WBTC_BADGER = 0  # TBD
+AURA_NEW_PID_WBTC_DIGG_GRAVI = 0  # TBD
+
+
+def sim():
+    # ownership transfer
+    trops = GreatApeSafe(r.badger_wallets.treasury_ops_multisig)
+
+    # avatars
+    aura_avatar = trops.contract(r.avatars.aura)
+    convex_avatar = trops.contract(r.avatars.convex)
+
+    aura_avatar.transferOwnership(r.badger_wallets.treasury_vault_multisig)
+    convex_avatar.transferOwnership(r.badger_wallets.treasury_vault_multisig)
+
+    main(True, True, True)
+
+
+def main(aura_migration=True, convex_migration=True, treasury_wd=True):
+    vault = GreatApeSafe(r.badger_wallets.treasury_vault_multisig)
+    vault.init_aura()
+
+    # avatars
+    aura_avatar = vault.contract(r.avatars.aura)
+    convex_avatar = vault.contract(r.avatars.convex)
+
+    # balancer bpts
+    bpt_badgerwbtc = vault.contract(r.balancer.B_20_BTC_80_BADGER)
+    bpt_badgerreth = vault.contract(r.balancer.B_50_BADGER_50_RETH)
+    bpt_wbtcdigggravi = vault.contract(r.balancer.bpt_40wbtc_40digg_20graviaura)
+
+    # convex private vault
+    private_vault = vault.contract(r.convex.frax.private_vaults.badger_fraxbp)
+
+    # convex wrap lp
+    wcvx_badger_fraxbp = vault.contract(r.convex.frax.wcvx_badger_fraxbp)
+
+    # snap
+    vault.take_snapshot(
+        tokens=[bpt_badgerwbtc, bpt_badgerreth, bpt_wbtcdigggravi, wcvx_badger_fraxbp]
+    )
+
+    if treasury_wd:
+        # aura wds
+        vault.aura.unstake_all_and_withdraw_all(bpt_badgerwbtc, claim=0)
+        vault.aura.unstake_all_and_withdraw_all(bpt_badgerreth, claim=0)
+        vault.aura.unstake_all_and_withdraw_all(bpt_wbtcdigggravi, claim=0)
+
+        # convex private wds
+        staking_contract = vault.contract(private_vault.stakingAddress())
+        kek_id = staking_contract.lockedStakes(private_vault, 0)[0]
+        private_vault.withdrawLocked(kek_id)
+
+        vault.print_snapshot()
+
+    if aura_migration:
+        total_assets = list(aura_avatar.totalAssets())
+
+        pids = list(aura_avatar.getPids())
+        if PID_AURA_EXCEPTION != 0:
```

Comment:
what is the purpose of this? checking a constant versus a hardcoded value?

Reply:
as touched on our meeting, modify lang and add onchain checks on 2945209862e13fab95358a879da13b7593e38d71